crates/orchestrator/src/database/mongodb/mod.rs

```diff
@@ -83,28 +83,28 @@ impl Database for MongoDb {
         Ok(self.get_job_collection().find_one(filter, None).await?)
     }
 
-    async fn update_job_status(&self, job: &JobItem, new_status: JobStatus) -> Result<()> {
-        let update = doc! {
-            "$set": {
-                "status": mongodb::bson::to_bson(&new_status)?,
-            }
-        };
-        self.update_job_optimistically(job, update).await?;
-        Ok(())
-    }
-
-    async fn update_external_id_and_status_and_metadata(
+    async fn update_job(
         &self,
         job: &JobItem,
         external_id: String,
         new_status: JobStatus,
         metadata: HashMap<String, String>,
     ) -> Result<()> {
+        let mut job_doc = bson::to_document(job)?;
+        job_doc.insert("external_id", external_id);
+        job_doc.insert("status", bson::to_bson(&new_status)?);
+        job_doc.insert("metadata", bson::to_document(&metadata)?);
```

Comment:
can we just take the job item only because that will have all the other stuff updated already

Reply:
Updated!